### PR TITLE
Added makepkg layer for editing Arch Linux PKGBUILD files

### PR DESCRIPTION
**Pull request by svenihoney**
This layer provides syntax highlighting and a couple of functions for [PKGBUILD](https://wiki.archlinux.org/index.php/PKGBUILD|PKGBUILD) files used by `makepkg` on [Arch Linux](https://www.archlinux.org).

### CODE REVIEW

File: layers/+tools/makepkg/packages.el [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+;; Salt mode URL: https://github.com/juergenhoetzel/pkgbuild-mode
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+(setq makepkg-packages '(pkgbuild-mode))
+
+(defun makepkg/init-pkgbuild-mode ()
+  (use-package pkgbuild-mode
+    :mode ("PKGBUILD\\'" . pkgbuild-mode)
```
Comment (Stebalien) [inline layers/+tools/makepkg/packages.el]: This should be `/PKGBUILD\\'"`.

File: layers/auto-layer.el [hunk 2]
```diff
@@ -82,6 +82,8 @@
 
 (configuration-layer/lazy-install 'php :extensions '("\\(\\.amk\\'\\|/Amkfile\\'\\|\\.phtml\\'\\|\\.php[s345t]?\\'\\|[^/]\\.\\(module\\|test\\|install\\|profile\\|tpl\\.php\\|theme\\|inc\\)\\'\\|\\.php\\'\\)" php-mode))
 
+(configuration-layer/lazy-install 'makepkg :extensions '("PKGBUILD\\'" pkgbuild-mode))
```
Comment (Stebalien) [inline layers/auto-layer.el]: This should be `/PKGBUILD\\'"`.

**Comment by svenihoney** [comment]
Corrected the file regex as suggested

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. You can safely delete your branch.

I moved it to the new `major-modes` layer since it was so small.

**Pull request closed without merging**